src/diem/testing/miniwallet/config.py

```diff
@@ -90,7 +90,9 @@ def serve() -> None:
 
     def start(self, client: jsonrpc.Client) -> threading.Thread:
         self.setup_account(client)
-        return self.serve(client)
+        t = self.serve(client)
+        utils.wait_for_port(self.server_conf.port)
```

Comment:
if the client binds to a nonlocal IP... will this still work?

Reply:
no matter what is ip, as server and client at same machine, you can always access the port by localhost, right?

Reply:
if you bind to a host that's not localhost, it will not listen to requests from localhost
```
import waitress, falcon

class Handler:
    def on_get(self, resp):
        resp.status = falcon.HTTP_200
        resp.body = "HELLO"

api = falcon.API()
api.add_route("/", Handler)

waitress.serve(
    api,
    host="192.168.1.28",
    port="7732",
)
```

![image](https://user-images.githubusercontent.com/1482859/110989144-4d12b680-8326-11eb-8463-89bbd13dd4d3.png)

![image](https://user-images.githubusercontent.com/1482859/110989177-59970f00-8326-11eb-91d7-0f49aeb64b5e.png)

Reply:
fixed by adding host to wait